kinto_client/importer.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+import argparse
+import getpass
+import os
+from six import iteritems
+
+from kinto_client import Client
+from kinto_client.exceptions import KintoException
+from kinto_client.logging import logging, global_logger, command_log
+
+
+def is_same_record(fields, one, two):
+    for key in fields:
+        if one.get(key) != two.get(key):
+            return False
+    return True
+
+
+class KintoImporter(object):
+    # Default configure parser values
+    include_all_default_parameters = False
+    include_remote_server = None
+    include_authentication = None
+    include_files = None
+    include_verbosity = None
+
+    # Remote Permissions
+    bucket_permissions = None
+    collection_permissions = None
+
+    # Default values
+    default_host = "http://localhost:8888/v1"
+    default_bucket = None
+    default_collection = None
+    default_auth = None
+    default_files = None
+
+    # Sync flags
+    create = True
+    update = True
+    delete = True
+
+    def __init__(self, logger=None, arguments=None,
+                 local_client=None, remote_client=None,
+                 *args, **kwargs):
+        if not hasattr(self, 'record_fields'):
+            raise ValueError(
+                "%r: record_fields attribute is not defined." % self)
+
+        self.logger = logger or global_logger
+
+        self._local_records = None
+        self._remote_records = None
+
+        parser = self.configure_parser(*args, **kwargs)
+        self.args = self.get_arguments(parser, args=arguments)
+        self.setup_logger()
+        self.local_client = self.setup_local_client(local_client)
+        self.remote_client = self.setup_remote_client(remote_client)
+
+    def configure_parser(self, parser=None,
+                         include_all_default_parameters=None,
+                         include_remote_server=None,
+                         include_authentication=None,
+                         include_files=None,
+                         include_verbosity=None,
+                         *args, **kwargs):
+        """Return an argparse.ArgumentParser pre-configured object.
+        :param argparse.ArgumentParser parser:
+            The parser to configure. Set to None it will create a new parser.
+
+        :param boolean include_all_default_parameters:
+            Configure the default for parameters defined below.
+            It is False by default. I can also be defined as a class attribute.
```

Comment:
I find this a bit overkill. Do we have use-cases where we don't want to allow server host or authentication?